js/filter-set.js

```diff
@@ -64,9 +63,9 @@ FilterSet.prototype.clear = function() {
   });
 };
 
-FilterSet.prototype.handleTagRemove = function(e) {
-  var $input = this.$body.find('#' + e.key);
-  var type = $input.attr('type');
+FilterSet.prototype.handleTagRemove = function(e, opts) {
+  var $input = this.$body.find('#' + opts.key);
+  var type = $input.attr('type') || 'text';
```

Comment:
Because "text" is the default type on an `<input>`. If we don't specify a default and ever use a plain `<input>` without a `type` attribute, this code will fail. We could drop the default if we want to require that all `<input>` elements in a `FilterSet` specify a `type`.